client/jetpack-connect/utils.js

```diff
@@ -14,11 +14,13 @@ import { head, includes, isEmpty, split } from 'lodash';
 import { addQueryArgs, untrailingslashit } from 'lib/route';
 
 export function authQueryTransformer( queryObject ) {
+	const truthy = [ 'true', 'True', 'TRUE', '1' ];
 	return {
 		// Required
 		clientId: parseInt( queryObject.client_id, 10 ),
-		closeWindowAfterLogin: !! queryObject.close_window_after_login,
+		closeWindowAfterLogin: -1 !== truthy.indexOf( queryObject.close_window_after_login ),
 		homeUrl: queryObject.home_url,
+		isPopup: -1 !== truthy.indexOf( queryObject.is_popup ),
```

Comment:
You could use ES6 version instead: `truthy.includes( queryObject.is_popup )` but I'm curious why we can't rely URL argument always be the same?

Reply:
good call. yeah, i suppose we can enforce it to always be 1

Reply:
addressd in 5bc0e3e017f3138aac9245845ef4548c678eba40